### PR TITLE
Topic algorithms

### PR DESCRIPTION
**Pull request by coppers-j**

### CODE REVIEW

File: image/equalize_adapthist_44312042/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Contrast Limited Adaptive Histogram Equalization (CLAHE)
```
Comment (shakes76) [inline image/equalize_adapthist_44312042/README.md]: Too minimal. Need info such as what it does, dependencies and example outputs/inputs

File: image/equalize_adapthist_44312042/equalize_adapthist.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
+"""
+COMP3710 Assignment 3
```
Comment (shakes76) [inline image/equalize_adapthist_44312042/equalize_adapthist.py]: Good header block

File: image/equalize_adapthist_44312042/equalize_adapthist.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    y, idx, count = tf.unique_with_counts(img)
+
+    for i in range(nbins):
+        if i==nbins-1:
+            mask = (img <= bins[i+1]) 
+            lim = tf.boolean_mask(img, mask)
+
+        else:
+            mask = (img < bins[i+1]) 
+            lim = tf.boolean_mask(img, mask)
```
Comment (shakes76) [inline image/equalize_adapthist_44312042/equalize_adapthist.py]: Good to add inline comments even if original code didn't have them

File: image/histogram_44312042/equalize_adapthist.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    
+    y, idx, count = tf.unique_with_counts(img)
+
+    for i in range(nbins):
+        if i==nbins-1:
+            mask = (img <= bins[i+1]) 
+            lim = tf.boolean_mask(img, mask)
+
+        else:
+            mask = (img < bins[i+1])
```
Comment (shakes76) [inline image/histogram_44312042/equalize_adapthist.py]: Good to add inline comments even if original code didn't have them

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme sparse
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove unnecessary files.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

File: image/equalize_adapthist_44312042/equalize_adapthist.py (hunk 3 above)
Comment (coppers-j) [reply image/equalize_adapthist_44312042/equalize_adapthist.py]: Added in latest commit

File: image/histogram_44312042/equalize_adapthist.py (hunk 4 above)
Comment (coppers-j) [reply image/histogram_44312042/equalize_adapthist.py]: Added in latest commit

File: image/equalize_adapthist_44312042/README.md (hunk 1 above)
Comment (coppers-j) [reply image/equalize_adapthist_44312042/README.md]: Updated with more info in latest commit

**Comment by coppers-j** [comment]
Removed unnecessary files and tidied up comments and README as requested

**Pull request merged**